app/services/fuzzySearchService.js

```diff
@@ -15,27 +17,24 @@ const fuseSearchOptions = {
 const getPaymentData = async ({ searchString, limit, offset, sortBy, filterBy, action }) => {
   if (!searchString) throw new Error('Empty search content')
 
-  const searchResults = await filterAndSearch(searchString, filterBy)
-  if (!searchResults.length) {
-    return { count: 0, rows: [] }
-  }
-
-  let results = getSortedResults(searchResults, sortBy)
-  if (action !== 'download') {
-    results = results.slice(offset, parseInt(offset) + parseInt(limit))
+  const searchResults = await search(searchString)
+  const filteredResults = applyFiltersAndGroupByPayee(searchResults, filterBy)
+  if (!filteredResults.length) {
+    return { count: 0, rows: [], filterOptions: {} }
```

Comment:
filterOptions represent filters which are valid for the results being returned. As there are no results, we don't send filterOptions back. The FE service will display all filters options as a result instead of filtering them based on the results.